### PR TITLE
LPT implemented

### PR DESCRIPTION
**Pull request by damonge**
LPT bias implemented

### CODE REVIEW

File: ClLike/cl_like/ept.py [hunk 1]
```diff
@@ -131,12 +132,55 @@ def _get_ia_bias(self, pk):
...
+        # k2k2 -> k^4 <d*d>, not provided
+        # When not provided, this function just returns `alt`
+
+        kind = kind.replace('m', 'd1')
+
+        if kind in self.pk2d_computed:
+            return self.pk2d_computed[kind]
+
+        if kind == 'd1d1':
+            return pnl
```
Comment (carlosggarcia) [inline ClLike/cl_like/ept.py:175]: where's `pnl` defined?

File: ClLike/cl_like/ept.py [hunk 2]
```diff
@@ -153,449 +197,23 @@ def get_pk(self, kind, pgrad=None, cosmo=None, sub_lowk=False):
...
-                'd2s2': 0.25,
-                's2s2': 0.25}
+        pfac = {'d1d2': 0.5, # d^2
+                'd1s2': 0.5, # s^2
+                'd2d2': 0.25, # d^2, d^2
+                'd2s2': 0.25, # d^2, s^2
+                's2s2': 0.25} # s^2, s^2
+
+        if kind not in inds:
+            return alt
```
Comment (carlosggarcia) [inline ClLike/cl_like/ept.py:207]: and `alt`?

File: ClLike/cl_like/ept.py [hunk 3]
```diff
@@ -131,12 +132,55 @@ def _get_ia_bias(self, pk):
...
+        # d1k2 -> k^2 <d*d> (with <d*d> as pnl)
+        # d2d2 -> <d^2/2*d^2/2>
+        # d2s2 -> <d^2/2*s^2/2>
+        # d2k2 -> k^2 <d*d^2/2>, not provided
+        # s2s2 -> <s^2/2*s^2/2>
+        # s2k2 -> k^2 <d*s^2/2>, not provided
+        # k2k2 -> k^4 <d*d>, not provided
+        # When not provided, this function just returns `alt`
+
+        kind = kind.replace('m', 'd1')
```
Comment (carlosggarcia) [inline ClLike/cl_like/ept.py:169]: What do you expect `kind` to be as input?

File: ClLike/cl_like/power_spectrum.py [hunk 4]
```diff
@@ -127,41 +127,35 @@ def _get_pk_data(self, cosmo):
...
-            pkd['pk_s2d1'] = pkd['pk_d1s2']
-            pkd['pk_s2d2'] = pkd['pk_d2s2']
-            pkd['pk_s2s2'] = ptc.get_pk('s2s2')
-            pkd['pk_s2k2'] = None
-            pkd['pk_k2m'] = pkd['pk_mk2']
-            pkd['pk_k2d1'] = pkd['pk_d1k2']
-            pkd['pk_k2d2'] = pkd['pk_d2k2']
-            pkd['pk_k2s2'] = pkd['pk_s2k2']
-            pkd['pk_k2k2'] = None
+            operators = ['m', 'd1', 'd2', 's2', 'k2']
```
Comment (carlosggarcia) [inline ClLike/cl_like/power_spectrum.py:143]: Nice!

File: ClLike/cl_like/limber.py [hunk 5]
```diff
@@ -92,9 +94,11 @@ def _get_tracers(self, cosmo, **pars):
                 z = dndz[0]
                 oz = np.ones_like(z)
 
-                t0 = None
                 tr = ccl.NumberCountsTracer(cosmo, dndz=dndz, bias=(z, oz),
                                             has_rsd=False)
+                t0 = None
```
Comment (carlosggarcia) [inline ClLike/cl_like/limber.py:99]: Can you add a comment here saying what t0, t1, tn, etc are? I know now what they are (t0 == unbiased tracer, right?). But we might forget it in a few months

**Review by carlosggarcia — CHANGES_REQUESTED** [review]
Looks good to me. I just left few minor comments.

File: ClLike/cl_like/ept.py (hunk 1 above)
Comment (damonge) [reply ClLike/cl_like/ept.py:175]: it's an argument of this function

File: ClLike/cl_like/ept.py (hunk 3 above)
Comment (carlosggarcia) [reply ClLike/cl_like/ept.py:169]: so `kind` is what's on the left of the arrow?

File: ClLike/cl_like/ept.py (hunk 1 above)
Comment (carlosggarcia) [reply ClLike/cl_like/ept.py:175]: (facepalm)

**Comment by carlosggarcia** [comment]
I forgot to ask for it before. @damonge, could you add the sacc file and test file you're using to test these? I will work on unifying the tests in a new PR

File: ClLike/cl_like/ept.py (hunk 3 above)
Comment (damonge) [reply ClLike/cl_like/ept.py:169]: yep

File: ClLike/cl_like/limber.py (hunk 5 above)
Comment (damonge) [reply ClLike/cl_like/limber.py:100]: Done

**Comment by damonge** [comment]
Just added all the test material without commenting anything at all. I don't think they're very useful as they are.

**Review by carlosggarcia — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**